### DISCUSSION TITLE
How to print learning rate

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Lightning-AI/pytorch-lightning#8492] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Here is asnippet of code
```
  def configure_optimizers(self):
    opt=torch.optim.AdamW(self.parameters(), lr=self.lr,weight_decay=1e-5)
    scheduler = ReduceLROnPlateau(opt,factor=0.5,patience=10,mode="min",verbose=True)
    return {'optimizer': opt,'lr_scheduler':scheduler,'monitor':'val/loss'}

  def train_dataloader(self):
    ds = Dataset(data=train_files, transform=train_transforms)
    loader=DataLoader(ds, batch_size=self.bs, shuffle=True,num_workers=self.numworker)
    return loader

  def training_step(self,batch,batch_idx):
    image,segment=batch["image"], batch["label"]
    out=self(image)
    loss=self.criterion(out,segment)
    self.log('train/loss', loss, on_epoch=False,prog_bar=True)
    return loss

  def val_dataloader(self):
    ds = Dataset(data=train_files, transform=val_transforms)
    loader=DataLoader(ds, batch_size=1, shuffle=False,num_workers=self.numworker)
    return loader
    
  def validation_step(self,batch,batch_idx):
    image,segment=batch["image"], batch["label"]
    outputs=self(image)
    loss=self.criterion(outputs,segment)
    self.log('val/dice', dice, on_epoch=True,prog_bar=True)
    return loss
  ```

Here is trainer
```

model = OurModel()
lr_monitor = LearningRateMonitor(logging_interval='epoch')
logger = TensorBoardLogger("logs", name="my_logs")
checkpoint_callback = ModelCheckpoint(monitor='val/loss',dirpath='/content/drive/MyDrive',
                                      filename='unetr' )
trainer = Trainer(max_epochs=500, auto_lr_find=False, auto_scale_batch_size=False,
                  logger=logger,
                  progress_bar_refresh_rate=120,
                  callbacks=[checkpoint_callback,lr_monitor ]
                  )
```
But I am not getting the learning rate displayed.
Currently the display is some thing like this
```[23:36<50:45, 11.80s/it, loss=0.813, v_num=5, val/dice=0.181, val/loss=0.811, train/loss=0.811, train/dice=0.198]```

**Comment by carmocca** [comment]
The `LearningRateMonitor` works by logging to a `logger`, not printing to stdout or to the progress bar.

To print to stdout, you could subclass and create your custom implementation, for example, this one:

https://github.com/carmocca/PyLaia/blob/master/laia/callbacks/learning_rate.py

To print to the progress bar, you can extract the learning rate manually in your `LightningModule` and call `self.log('lr', lr, prog_bar=True)`
